src/plots/smith/layout_attributes.js

```diff
@@ -0,0 +1,206 @@
+    // but realaxis lines are straight from circle center to outer bound
+    axisLineGridAttr,
+    axisTickAttrs
+);
+
+var angularAxisAttrs = {
+    visible: extendFlat({}, axesAttrs.visible, {dflt: true}),
+
+    direction: {
+        valType: 'enumerated',
+        values: ['counterclockwise', 'clockwise'],
+        dflt: 'counterclockwise',
+        editType: 'calc',
+        description: [
+            'Sets the direction corresponding to positive angles.'
+        ].join(' ')
+    },
+
+    rotation: {
+        valType: 'angle',
+        editType: 'calc',
+        description: [
+            'Sets that start position (in degrees) of the angular axis',
+            'By default, smith subplots with `direction` set to *counterclockwise*',
+            'get a `rotation` of *0*',
+            'which corresponds to due East (like what mathematicians prefer).',
+            'In turn, smith with `direction` set to *clockwise* get a rotation of *90*',
+            'which corresponds to due North (like on a compass),'
+        ].join(' ')
+    },
+
+    hoverformat: axesAttrs.hoverformat,
+
+    uirevision: {
+        valType: 'any',
+        editType: 'none',
+        description: [
+            'Controls persistence of user-driven changes in axis `rotation`.',
+            'Defaults to `smith<N>.uirevision`.'
+        ].join(' ')
+    },
+
+    editType: 'calc'
+};
+
+extendFlat(
+    angularAxisAttrs,
+
+    // N.B. angular grid lines are straight lines from circle center to outer bound
+    // the angular line is circular bounding the smith plot area.
+    axisLineGridAttr,
+
+    // N.B. ticksuffix defaults to '°' for angular axes with `thetaunit: 'degrees'`
+    axisTickAttrs
+);
+
+module.exports = {
+    // TODO for x/y/zoom system for paper-based zooming:
+    // x: {},
+    // y: {},
+    // zoom: {},
+
+    domain: domainAttrs({name: 'smith', editType: 'plot'}),
+
+    sector: {
+        valType: 'info_array',
+        items: [
+            {valType: 'number', editType: 'plot'},
+            {valType: 'number', editType: 'plot'}
+        ],
+        dflt: [0, 360],
+        editType: 'plot',
+        description: [
+            'Sets angular span of this smith subplot with two angles (in degrees).',
+            'Sector are assumed to be spanned in the counterclockwise direction',
+            'with *0* corresponding to rightmost limit of the smith subplot.'
+        ].join(' ')
+    },
+    hole: {
+        valType: 'number',
+        min: 0,
+        max: 1,
+        dflt: 0,
+        editType: 'plot',
+        description: [
+            'Sets the fraction of the radius to cut out of the smith subplot.'
+        ].join(' ')
+    },
+
+    bgcolor: {
+        valType: 'color',
+        editType: 'plot',
+        dflt: colorAttrs.background,
+        description: 'Set the background color of the subplot'
+    },
+
+    realaxis: radialAxisAttrs,
+    imaginaryaxis: angularAxisAttrs,
+
+    gridshape: {
```

Comment:
`gridshape` is supported?

Reply:
021d7086a5e40b077a306e4a8ec1019422d1829a